### PR TITLE
CLOUDSTACK-8728: Testcase to Verify if VRs IP 

### PR DESCRIPTION
**Pull request by pritisarap12**
Testcase to check if VR's IP changes if it is destroyed and re-created in basic zone.

### CODE REVIEW

**Comment by pritisarap12** [comment]
Testcase result:

Check if Virtual Router has same IP after network restart or not. ... === TestName: test_01_vr_has_same_ip | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 245.730s

OK

**Comment by pritisarap12** [comment]
Merged the testcase in test_routers.py file.

**Comment by pritisarap12** [comment]
Testcase result:

Check if Virtual Router has same IP after network restart or not. ... === TestName: test_10_vr_has_same_ip | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 294.769s

OK

**Comment by bhaisaab** [comment]
LGTM

File: test/integration/smoke/test_routers.py [hunk 1]
```diff
@@ -796,3 +830,122 @@ def test_09_reboot_router(self):
...
+        host_id = list_virtual_machine_response[0].hostid
+        list_host_response = list_hosts(self.apiclient, id=host_id)
+        status = validateList(list_host_response)
+        self.assertEqual(
+            status[0],
+            PASS,
+            "Host List Validation Failed due to %s" %
+            status[2])
+
+        self.services['hosts'][self.hypervisor.lower()]['password'] = "xenroot"
```
Comment (sanju1010) [inline test/integration/smoke/test_routers.py]: Can you please remove the hardcoded password for the host and see if you can get from the config file?

File: test/integration/smoke/test_routers.py [hunk 2]
```diff
@@ -796,3 +830,122 @@ def test_09_reboot_router(self):
...
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype
+        )
+        self.check_virtual_machine_accessibility(vm_2)
+        ssh_object = self.try_ssh_to_host()
+        inet_address_after_restart = self.get_inet_address(ssh_object)
```
Comment (sanju1010) [inline test/integration/smoke/test_routers.py]: After network restart we are suppose to verify the VR's IP address not the SSVM ip address. Can you please double check this test?

File: test/integration/smoke/test_routers.py [hunk 3]
```diff
@@ -796,3 +830,122 @@ def test_09_reboot_router(self):
...
+        status = validateList(list_ssvm_response)
+        self.assertEqual(
+            status[0],
+            PASS,
+            "SSVM List Validation Failed due to %s" %
+            status[2])
+
+        command = "ssh  -o StrictHostKeyChecking=no -i /root/.ssh/id_rsa.cloud %s -p 3922 'ifconfig eth0'" \
+            % list_ssvm_response[0].linklocalip
+        ssh_response = ssh_object.execute(command)
```
Comment (sanju1010) [inline test/integration/smoke/test_routers.py]: What are we trying to get here? linklocalip and "ifconfig eth0" would be same on SSVM. You may have to check this for Router VM. Please modify this method implementation.

File: test/integration/smoke/test_routers.py [hunk 4]
```diff
@@ -796,3 +830,122 @@ def test_09_reboot_router(self):
...
+        ssh_object = SshClient(
+            host=list_host_response[0].ipaddress,
+            port=self.services['natrule']['publicport'],
+            user=self.services['hosts'][self.hypervisor.lower()]['username'],
+            passwd=self.services['hosts'][self.hypervisor.lower()]['password']
+        )
+        hostname_from_host = ssh_object.execute("uname -n")
+        ssh_object.execute("ssh-keygen -R %s" % hostname_from_host[0])
+        ssh_object.execute("exit")
+        return ssh_object
```
Comment (sanju1010) [inline test/integration/smoke/test_routers.py]: What is the guarantee that  RouterVM and Guest vm will be on the same host? We have to the host id of the VR and  get the ssh object for that host.

**Comment by miguelaferreira** [comment]
Something strange with this PR. The second commit merges master into the PR

Was this an attempt to bring in new changes from master? If that was the case, then a rebase would have been the better option.

@pritisarap12 can you please remove the merge commit, and rebase instead?
Let me know if you would like me to help you with that.

**Comment by remibergsma** [comment]
@pritisarap12 Please squash the commits (get rid of the merge commit). We cannot merge it like this. Thanks!

**Comment by miguelaferreira** [comment]
@remi: do you mean squash to get rid of the merge commit? Wouldn't the resulting commit have 2 parents? If so, that's not the solution.

Or do you mean squash all but the merge commit?

**Comment by remibergsma** [comment]
@miguelaferreira If I recall correctly, `rebase -i` (where you'd usually squash stuff together and such) and doing nothing, will remove the merge commit already. It's better to say to rebase I guess ;-) End goal is indeed to keep only the first commit. Cc @pritisarap12

**Comment by miguelaferreira** [comment]
@remibergsma `rebase -i` and deleting the line of the merge commit will get you what you want. I see that squashing would too, but then the problem is in the generalisation that tends to happen in the next PR someone makes. People might thing, let me squash everything before someone asks me to do it ;)

**Comment by remibergsma** [comment]
@miguelaferreira OK, check.

**Comment by remibergsma** [comment]
@pritisarap12 Any update on this?

**Comment by pritisarap12** [comment]
Merged the testcase in the existing testcase test_03_restart_network_cleanup by adding support for basic setup.

**Comment by remibergsma** [comment]
@pritisarap12 Please squash the commits, as there are now 2 merge commits and I cannot proceed like this. Make sure it has a single commit. Ping me if you need help with this.

**Comment by prits09** [comment]
squashed the commits into single commit.

**Comment by remibergsma** [comment]
@prits09 Thanks! It seems Jenkins failed and I don't think it's you. Please force-push the commit again so the rests will re-run. After that, I'll run the test myself.

**Comment by pritisarap12** [comment]
@remibergsma:  I force pushed the commit and it is showing green signal now you can run the test.

**Comment by sanju1010** [comment]
LGTM !!

Test router internal basic zone ... SKIP: Marvin configuration has no host credentials to                            check router services
Test router internal advanced zone ... SKIP: Marvin configuration has no host credentials                            to check router services
Test restart network ... === TestName: test_03_restart_network_cleanup | Status : SUCCESS ===
ok
Test restart network without cleanup ... SKIP: Marvin configuration has no host credentials                            to check router services
Test router basic setup ... === TestName: test_05_router_basic | Status : SUCCESS ===
ok
Test router advanced setup ... === TestName: test_06_router_advanced | Status : SUCCESS ===
ok
Test stop router ... === TestName: test_07_stop_router | Status : SUCCESS ===
ok
Test start router ... === TestName: test_08_start_router | Status : SUCCESS ===
ok
Test reboot router ... === TestName: test_09_reboot_router | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 9 tests in 287.782s

OK (SKIP=3)

**Pull request merged**